pyomo/contrib/doe/tests/test_fim_doe.py

```diff
@@ -25,68 +25,350 @@
 #  publicly, and to permit other to do so.
 #  ___________________________________________________________________________
 
+from pyomo.common.dependencies import numpy as np, numpy_available
 import pyomo.common.unittest as unittest
-from pyomo.contrib.doe import Measurements, Scenario_generator
+from pyomo.contrib.doe import (
+    MeasurementVariables,
+    DesignVariables,
+    ScenarioGenerator,
+    DesignOfExperiments,
+    VariablesWithIndices,
+)
+from pyomo.contrib.doe.examples.reactor_kinetics import create_model, disc_for_measure
+
+
+class TestMeasurementError(unittest.TestCase):
+    def test(self):
+        t_control = [0, 0.125, 0.25, 0.375, 0.5, 0.625, 0.75, 0.875, 1]
+        variable_name = "C"
+        indices = {0: ['CA', 'CB', 'CC'], 1: t_control}
+        # measurement object
+        measurements = MeasurementVariables()
+        # if time index is not in indices, an value error is thrown.
+        self.assertRaises(
+            ValueError,
+            MeasurementVariables.add_variables,
+            measurements,
+            variable_name,
+            indices=indices,
+            time_index_position=2,
+        )
```

Comment:
Can you switch this to use a context manager (example below)? It makes it much easier to decipher the code you expect to throw an error.

    with self.assertRaises(ValueError):
        # Code that should raise a ValueError